oracle-plugin/src/main/java/io/cdap/plugin/oracle/OracleConstants.java

```diff
@@ -29,37 +29,66 @@ private OracleConstants() {
   public static final String DEFAULT_ROW_PREFETCH = "defaultRowPrefetch";
   public static final String SERVICE_CONNECTION_TYPE = "service";
   public static final String CONNECTION_TYPE = "connectionType";
   public static final String ROLE = "role";
   public static final String NAME_DATABASE = "database";
-  public static final String TNS_CONNECTION_TYPE = "TNS";
+  public static final String TNS_CONNECTION_TYPE = "tns";
   public static final String TRANSACTION_ISOLATION_LEVEL = "transactionIsolationLevel";
+  public static final String USE_SSL = "useSSL";
 
   /**
-   * Returns the Connection String for the given ConnectionType.
+   * Constructs the Oracle connection string based on the provided connection type, host, port, and database.
+   * If SSL is enabled, the connection protocol will be "tcps" instead of "tcp".
    *
    * @param connectionType TNS/Service/SID
    * @param host Host name of the oracle server
    * @param port Port of the oracle server
    * @param database Database to connect to
-   * @return Connection String based on the given ConnectionType
+   * @param useSSL Whether SSL/TLS is required(YES/NO)
+   * @return Connection String based on the given parameters and connection type.
    */
   public static String getConnectionString(String connectionType,
                                            @Nullable String host,
                                            @Nullable int port,
-                                           String database) {
-    if (OracleConstants.TNS_CONNECTION_TYPE.equalsIgnoreCase(connectionType)) {
-      return String.format(OracleConstants.ORACLE_CONNECTION_STRING_TNS_FORMAT, database);
+                                           String database,
+                                           @Nullable Boolean useSSL) {
+    // Use protocol as "tcps" when SSL is requested or else use "tcp".
+    String connectionProtocol;
+    boolean isSSLEnabled = false;
+    if (useSSL != null && useSSL) {
+      connectionProtocol = "tcps";
+      isSSLEnabled = true;
+    } else {
+      connectionProtocol = "tcp";
     }
-    if (OracleConstants.SERVICE_CONNECTION_TYPE.equalsIgnoreCase(connectionType)) {
-      return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SERVICE_NAME_FORMAT,
-          host, port, database);
+
+    switch (connectionType.toLowerCase()) {
+      case OracleConstants.TNS_CONNECTION_TYPE:
+        // TNS connection doesn't require protocol
+        return String.format(OracleConstants.ORACLE_CONNECTION_STRING_TNS_FORMAT, database);
+      case OracleConstants.SERVICE_CONNECTION_TYPE:
+        if (isSSLEnabled) {
+          return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SERVICE_NAME_FORMAT_WITH_PROTOCOL,
+              connectionProtocol, host, port, database);
+        }
+        return String.format(OracleConstants.ORACLE_CONNECTION_STRING_SERVICE_NAME_FORMAT,
+            host, port, database);
```

Comment:
can we have two methods which takes `host, port, database & connectionProtocol` `getConnectionStringWithServic` & `getConnectionStringWithSID`?

Reply:
Added separate methods to get the different type of connection strings. [7dc9a6c7f9d409c8f3debe71085df9d10ecaeb0a]